### DISCUSSION TITLE
Ensemble statistics (percentiles)

### DISCUSSION THREAD

**Discussion by TarekAMZ** [pmlmodelling/nctoolkit#120] [Q&A]
Hi,
I was wondering if it is possible to calculate the ensemble percentiles across all timesteps, e.g., similar to calculating the ensemble mean or max using ignore_time = True like the example below from the documentation.

data = nc.open_data(file_list)
data.ensemble_max(ignore_time = True)

I appreciate the help.

**Comment by robertjwilson** [comment]
Thanks for using the package @TarekAMZ.

Right now, the functionality does not exist. I will raise an issue to add this.

Depending on how your data is structured, the following could work

    data = nc.open_data(file_list)
    data.merge("time")
    data.tpercentile(....)

Under the hood, nctoolkit will use CDO's `mergetime` operator to combine files in this case. If you have duplicates, it  will end up duplicating times if they appear across multiple files. I think it _should_ work in most cases. Though at times you can have an ensemble where time is defined in a totally incompatible way so the `mergetime` operator doesn't work. I'd try it and check for warnings/errors which ought to be self-explanatory.

Alternatively, you could try this:

    data = nc.open_data(file_list)
    data.cdo_command("cat", ensemble = True)
    data.tpercentile(....)

In most cases, I suspect either approach will work. It's just a question of whether the file structure is such that they can be merged.
